mission/mission.proto

```diff
@@ -6,25 +6,63 @@ option java_package = "io.dronecore.mission";
 option java_outer_classname = "MissionProto";
 
 service MissionService {
-    rpc SendMission(SendMissionRequest) returns(MissionResult) {}
-    rpc StartMission(StartMissionRequest) returns(MissionResult) {}
+    rpc UploadMission(UploadMissionRequest) returns(UploadMissionResponse) {}
+    rpc DownloadMission(DownloadMissionRequest) returns(DownloadMissionResponse) {}
+    rpc StartMission(StartMissionRequest) returns(StartMissionResponse) {}
+    rpc PauseMission(PauseMissionRequest) returns(PauseMissionResponse) {}
+    rpc SetCurrentMissionItemId(SetCurrentMissionItemIdRequest) returns(SetCurrentMissionItemIdResponse) {}
+    rpc GetCurrentMissionItemId(GetCurrentMissionItemIdRequest) returns(GetCurrentMissionItemIdResponse) {}
+    rpc GetMissionSize(GetMissionSizeRequest) returns(GetMissionSizeResponse) {}
+    rpc SubscribeMissionProgress(SubscribeMissionProgressRequest) returns(stream MissionProgressResponse) {}
```

Comment:
We can add ClearMission to delete all mission items at once

Reply:
It should be supported from DroneCore C++ first :)

Reply:
It is part of Mavlink message specs

Reply:
True, but we need a hook in DroneCore C++ interface.

Reply:
Good points. Nothing prevents us from extending the protocol later. So as soon as we have a "ClearMission" implementation on the DroneCore libs, we can add it to the protocol :-).